### DISCUSSION TITLE
hyprland

### DISCUSSION THREAD

**Discussion by aloispichler** [davatorium/rofi#1921] [Q&A] [answered]
I am running rofi on i3 and hyprland.
While everything’s okay in i3, `rofi -show window` lists only a fraction of open windows.
So, I think I am missing something – but what?

**Comment by carnager** [comment, marked as answer]
rofi is not Wayland native, so it only displays X applications

**Reply by aloispichler** [reply]
I see, thank you. Is there a way or a script which includes wayland apps? `hyprctl clients` provides the list, so probably this should be simple in rofi?

**Reply by aloispichler** [reply]
I’ll try this, thank you. I was hoping for a more integrated solution, though… Thanks again

**Comment by carnager** [comment]
One example I could find quickly: https://github.com/Hyppothalamus/hyprland-rofi-window-switcher

**Comment by mcepl** [comment]
Use https://github.com/lbonn/rofi
